cart/application/cartService.go

```diff
@@ -896,6 +911,16 @@ func (cs *CartService) AdjustItemsToRestrictedQty(ctx context.Context, session *
 				return nil, err
 			}
 
+			if product.Type() == productDomain.TypeConfigurable {
```

Comment:
Smells like duplicate code. Maybe move to separate function?

Reply:
Moved to a separate function, please have a look.